core/src/main/scala/io/qbeast/spark/index/SparkPlanAnalyzer.scala

```diff
@@ -0,0 +1,109 @@
+package io.qbeast.spark.index
+
+import org.apache.spark.sql.catalyst.expressions.Alias
+import org.apache.spark.sql.catalyst.expressions.Expression
+import org.apache.spark.sql.catalyst.plans.logical.Filter
+import org.apache.spark.sql.catalyst.plans.logical.LocalLimit
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.catalyst.plans.logical.Project
+import org.apache.spark.sql.catalyst.plans.logical.Sample
+import org.apache.spark.sql.catalyst.plans.logical.Sort
+import org.apache.spark.sql.DataFrame
+
+import scala.collection.convert.ImplicitConversions.`collection asJava`
+
+object SparkPlanAnalyzer {
```

Comment:
Better use a `trait` and have `DataAnalyzer` extend it.

Reply:
sure!